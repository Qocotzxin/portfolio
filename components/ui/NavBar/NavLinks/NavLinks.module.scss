.NavLinks {
  display: none;
  align-items: center;
  justify-content: center;
  list-style: none;

  @media only screen and (min-width: 48em) {
    display: flex;
  }

  li {
    margin-right: var(--chakra-space-2);
    margin-left: var(--chakra-space-2);
  }
}

.FloatingMenu {
  display: block;

  @media only screen and (min-width: 48em) {
    display: none;
  }
}
