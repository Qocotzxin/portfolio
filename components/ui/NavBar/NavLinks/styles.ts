import styles from "./NavLinks.module.scss";

export const Classes = {
  navLinks: styles["NavLinks"],
  floatingMenu: styles["FloatingMenu"],
};
