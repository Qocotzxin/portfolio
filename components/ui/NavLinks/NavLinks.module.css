.link {
    text-shadow: 0 0 6px var(--chakra-colors-teal-100), 0 0 6px var(--chakra-colors-teal-100), 0px 0 6px var(--chakra-colors-blue-300), 0px 0 6px var(--chakra-colors-blue-300);
}