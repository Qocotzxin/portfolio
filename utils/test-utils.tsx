import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "contexts/hygraphContext";
import { Components, HygraphData } from "@models/hygraph";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
