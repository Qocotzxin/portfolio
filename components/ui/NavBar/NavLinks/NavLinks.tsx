import { useHygraphContext } from "@contexts/hygraphContext";
import { useAriaLabel } from "@hooks/useAriaLabel";
import { Hamburger } from "@icons";
import { Components } from "@models/hygraph";
import BaseLink from "@ui/BaseLink";
import FloatingMenu from "@ui/FloatingMenu";
import { removeForwardSlash } from "@utils/url";
import { useRouter } from "next/router";
import { FC } from "react";
import { Classes } from "./styles";

const NavLinks: FC = () => {
  const router = useRouter();
  const currentPath = removeForwardSlash(router.pathname);
  const { navLinks } = useHygraphContext();
  const ariaLabel = useAriaLabel({ component: Components.NavLinks });

  return (
    <>
      {/* Desktop */}
      <ul data-testid="NavLinks-desktop" className={Classes.navLinks}>
        {navLinks?.map((link) => (
          <li key={link.url}>
            <BaseLink activeRoute={currentPath} link={link} />
          </li>
        ))}
      </ul>
      {/* Mobile */}
      <div className={Classes.floatingMenu}>
        <FloatingMenu
          ariaLabel={ariaLabel?.content}
          ButtonIcon={<Hamburger />}
          data-testid="NavLinks-mobile"
        >
          {navLinks?.map((link) => (
            <BaseLink
              key={link.url}
              isMenuItem
              activeRoute={currentPath}
              link={link}
            />
          ))}
        </FloatingMenu>
      </div>
    </>
  );
};

export default NavLinks;
