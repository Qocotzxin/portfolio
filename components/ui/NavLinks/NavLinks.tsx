import { HStack, Link, List, ListItem, VisuallyHidden } from "@chakra-ui/react";
import { useHygraphContext } from "@context/hygraphContext";
import NextLink from "next/link";
import { useRouter } from "next/router";
import styles from "./NavLinks.module.css";

const NavLinks = () => {
  const router = useRouter();
  const { navLinks } = useHygraphContext();

  return (
    <List display="flex" alignItems="center" justifyContent="center">
      {navLinks?.map((link) => (
        <ListItem key={link.url} mx="2">
          <NextLink
            href={router.pathname === link.url ? "#main" : `/${link.url}`}
            passHref
          >
            <Link
              variant="base"
              order={link.order}
              aria-current={router.pathname === `/${link.url}`}
              transition="text-shadow 0.2s linear"
              className={router.pathname === `/${link.url}` ? styles.link : ""}
            >
              {link.text}
            </Link>
          </NextLink>
        </ListItem>
      ))}
    </List>
  );
};

export default NavLinks;
