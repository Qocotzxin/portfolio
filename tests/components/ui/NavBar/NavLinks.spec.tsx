import NavLinks from "@ui/NavBar/NavLinks";
import { render } from "@utils/test-utils";

describe("NavLinks", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<NavLinks />);

    expect(container).toMatchSnapshot();
  });
});
