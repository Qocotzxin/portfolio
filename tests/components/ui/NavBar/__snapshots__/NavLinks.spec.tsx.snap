// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks Should match snapshot. 1`] = `
.emotion-0 {
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-fast);
  transition-timing-function: var(--chakra-transition-easing-ease-out);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: inherit;
  font-family: var(--chakra-fonts-mono);
  font-size: var(--chakra-fontSizes-xs);
  font-weight: var(--chakra-fontWeights-bold);
  -webkit-transition: text-shadow 0.2s linear,color 0.2s linear;
  transition: text-shadow 0.2s linear,color 0.2s linear;
  position: relative;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  text-transform: capitalize;
}

.emotion-0:hover,
.emotion-0[data-hover] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover::before,
.emotion-0[data-hover]::before {
  -webkit-transform: scaleX(1);
  -moz-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.emotion-0:focus-visible,
.emotion-0[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-0::before {
  content: "";
  position: absolute;
  display: block;
  width: 100%;
  height: 2px;
  bottom: calc(var(--chakra-space-1) * -1);
  left: 0px;
  background-color: var(--chakra-colors-gray-800);
  -webkit-transform: scaleX(0);
  -moz-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 0.3s ease;
  transition: transform 0.3s ease;
}

.chakra-ui-dark .emotion-0:not([data-theme])::before,
[data-theme=dark] .emotion-0:not([data-theme])::before,
.emotion-0[data-theme=dark]::before {
  background-color: var(--chakra-colors-gray-50);
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-8);
  min-width: var(--chakra-sizes-8);
  font-size: var(--chakra-fontSizes-sm);
  -webkit-padding-start: var(--chakra-space-3);
  padding-inline-start: var(--chakra-space-3);
  -webkit-padding-end: var(--chakra-space-3);
  padding-inline-end: var(--chakra-space-3);
  color: var(--chakra-colors-blue-200);
  background: var(--chakra-colors-transparent);
  padding: 0px;
}

.emotion-2:focus-visible,
.emotion-2[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-2[disabled],
.emotion-2[aria-disabled=true],
.emotion-2[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-2:hover,
.emotion-2[data-hover] {
  background: rgba(144, 205, 244, 0.12);
}

.emotion-2:hover[disabled],
.emotion-2[data-hover][disabled],
.emotion-2:hover[aria-disabled=true],
.emotion-2[data-hover][aria-disabled=true],
.emotion-2:hover[data-disabled],
.emotion-2[data-hover][data-disabled] {
  background: initial;
}

.emotion-2:active,
.emotion-2[data-active] {
  background: rgba(144, 205, 244, 0.24);
}

.emotion-3 {
  width: 24px;
  height: 24px;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: currentColor;
}

.emotion-4 {
  z-index: 1;
}

.emotion-5 {
  outline: 2px solid transparent;
  outline-offset: 2px;
  background: var(--chakra-colors-gray-700);
  box-shadow: var(--chakra-shadows-dark-lg);
  color: inherit;
  min-width: var(--chakra-sizes-3xs);
  padding-top: var(--chakra-space-2);
  padding-bottom: var(--chakra-space-2);
  z-index: 1;
  border-radius: var(--chakra-radii-md);
  border-width: 1px;
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
}

.emotion-6 {
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-fast);
  transition-timing-function: var(--chakra-transition-easing-ease-out);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: inherit;
  font-family: var(--chakra-fonts-mono);
  font-size: var(--chakra-fontSizes-xs);
  font-weight: var(--chakra-fontWeights-bold);
  -webkit-transition: text-shadow 0.2s linear,color 0.2s linear;
  transition: text-shadow 0.2s linear,color 0.2s linear;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: start;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  outline: 2px solid transparent;
  outline-offset: 2px;
  padding-top: var(--chakra-space-1-5);
  padding-bottom: var(--chakra-space-1-5);
  -webkit-padding-start: var(--chakra-space-3);
  padding-inline-start: var(--chakra-space-3);
  -webkit-padding-end: var(--chakra-space-3);
  padding-inline-end: var(--chakra-space-3);
  transition-property: var(--chakra-transition-property-background);
  transition-duration: var(--chakra-transition-duration-ultra-fast);
  transition-timing-function: var(--chakra-transition-easing-ease-in);
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  text-transform: capitalize;
}

.emotion-6:hover,
.emotion-6[data-hover] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:hover::before,
.emotion-6[data-hover]::before {
  -webkit-transform: scaleX(1);
  -moz-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.emotion-6:focus-visible,
.emotion-6[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-6::before {
  content: "";
  position: absolute;
  display: block;
  width: 100%;
  height: 2px;
  bottom: calc(var(--chakra-space-1) * -1);
  left: 0px;
  background-color: var(--chakra-colors-gray-800);
  -webkit-transform: scaleX(0);
  -moz-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 0.3s ease;
  transition: transform 0.3s ease;
}

.chakra-ui-dark .emotion-6:not([data-theme])::before,
[data-theme=dark] .emotion-6:not([data-theme])::before,
.emotion-6[data-theme=dark]::before {
  background-color: var(--chakra-colors-gray-50);
}

.emotion-6:focus,
.emotion-6[data-focus] {
  background: var(--chakra-colors-whiteAlpha-100);
}

.emotion-6:active,
.emotion-6[data-active] {
  background: var(--chakra-colors-whiteAlpha-200);
}

.emotion-6[aria-expanded=true],
.emotion-6[data-expanded] {
  background: var(--chakra-colors-whiteAlpha-100);
}

.emotion-6[disabled],
.emotion-6[aria-disabled=true],
.emotion-6[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
}

<div>
  <ul
    class="NavLinks"
    data-testid="NavLinks-desktop"
  >
    <li>
      <a
        aria-current="false"
        class="chakra-link emotion-0"
        href="/test"
      >
        Test
      </a>
    </li>
  </ul>
  <div
    class="FloatingMenu"
  >
    <div
      class="emotion-1"
      data-testid="NavLinks-mobile"
    >
      <button
        aria-controls="menu-list-:r1:"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-button chakra-menu__menu-button emotion-2"
        data-testid="FloatingMenu-button"
        id="menu-button-:r1:"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon emotion-3"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16v2H4zm0 5h16v2H4zm0 5h16v2H4z"
          />
        </svg>
      </button>
      <div
        class="emotion-4"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list emotion-5"
          id="menu-list-:r1:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <a
            aria-current="false"
            class="chakra-link chakra-menu__menuitem emotion-6"
            data-index="0"
            href="/test"
            id="menu-list-:r1:-menuitem-:r2:"
            role="menuitem"
            tabindex="-1"
          >
            Test
          </a>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
